elastic/kibana
# [Drilldowns] Fix missing $store in filter state syncing edge case

## Summary

Part of https://github.com/elastic/kibana/issues/61230
Originally I put this fix into https://github.com/elastic/kibana/pull/60087, but as it has nothing to do with drilldowns, extracting into separate pr. 

In drilldown branch during navigating to dashboard filters were generated without `store` value explicitly set (this could theoretically happen in other places, as the value is optional).
In url we were setting those filters into either `_a` or `_g` query param, which makes it clear if filter is unpinned or pinned. 
After navigation `store` was add to the filter object, changing it's shape and state syncing utils run a redundant browser history update which broke back button. 

This pr change makes sure that data service state syncing utilities are ignoring update of internal filter's `$state` value if specific type of filters (pinned or unpinned) is being synced.  


### Testing

It is hard to test without drilldowns branch context: https://github.com/elastic/kibana/pull/60087
Functional test will be added together with drilldown https://github.com/elastic/kibana/pull/60087#discussion_r397242773

It is possible to "mimic" the navigation which will happen in drilldown: 

1. Create a dashboard "origin" 
2. Create a dashboard "destination". Get a url to this dashboard with some filter applied. remove `$store.state` part from the filter object in the link. 
3. Create a Markdown vis and and render the link to destination dashboard with modified filter there. Add Markdown vis to origin dashboard.
4. Go to original dashboard, click on a link. you should be navigated to destination dashboard with applied filter from the url. 
5. Press back button 

Expected behaviour: navigated back to origin dashboard
Current in master: back didn't work
